### PR TITLE
Sidebar Done

### PR DESCRIPTION
**Pull request by Snayak7643**

### CODE REVIEW

File: src/Components/Navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import { Link } from "react-router-dom";
+import { IoLogoDesignernews } from "react-icons/io";
+import { TiArrowBack } from "react-icons/ti";
+import { FaBars } from "react-icons/fa";
+
+const Navbar = () => {
+  const [show, setShow] = useState<boolean>(false);
+
+  const handleClick = () => {
+    setShow(!show);
```
Comment (Ishaan-flyfin) [inline src/Components/Navbar.tsx]: there is a way to do this
setShow(prevState => !prevState)
do read about it and let me know

File: src/App.tsx [hunk 2]
```diff
@@ -1,7 +1,44 @@
...
-  return <h1>Sidebar-modal</h1>;
+  const [open, setOpen] = useState<string>("");
+
+  const openModal = (name: string) => {
+    setOpen(name);
+  };
+
+  const closeModal = () => {
+    setOpen("");
+  };
```
Comment (Ishaan-flyfin) [inline src/App.tsx]: lets use custom hook for this
will sync on this today

File: src/App.tsx [hunk 3]
```diff
@@ -1,7 +1,44 @@
...
+import Contact from "./Pages/Contact";
+import Modal from "./Components/Modal";
+
+type ModalType = {
+  open: string;
+  openModal: (name: string) => void;
+  closeModal: () => void;
+};
+
+const ModalContext = createContext<ModalType>({} as ModalType);
```
Comment (Ishaan-flyfin) [inline src/App.tsx]: we can create a separate file for this

**Review by Ishaan-flyfin — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 3 above)
Comment (Snayak7643) [reply src/App.tsx]: created a separate contexts file

File: src/Components/Navbar.tsx (hunk 1 above)
Comment (Snayak7643) [reply src/Components/Navbar.tsx]: Done

File: src/App.tsx (hunk 2 above)
Comment (Snayak7643) [reply src/App.tsx]: created a new useModal hook

**Pull request merged**